include/SFML/Graphics/Rect.hpp

```diff
@@ -180,49 +236,36 @@ using FloatRect = Rect<float>;
 /// \class sf::Rect
 /// \ingroup graphics
 ///
-/// A rectangle is defined by its top-left corner and its size.
-/// It is a very simple class defined for convenience, so
-/// its member variables (left, top, width and height) are public
-/// and can be accessed directly, just like the vector classes
-/// (Vector2 and Vector3).
-///
-/// To keep things simple, sf::Rect doesn't define
-/// functions to emulate the properties that are not directly
-/// members (such as right, bottom, center, etc.), it rather
-/// only provides intersection functions.
+/// A rectangle is defined by its position and size
 ///
-/// sf::Rect uses the usual rules for its boundaries:
+/// sf::Rect uses the following rules for its boundaries:
 /// \li The left and top edges are included in the rectangle's area
-/// \li The right (left + width) and bottom (top + height) edges are excluded from the rectangle's area
+/// \li The right and bottom edges are excluded from the rectangle's area
 ///
-/// This means that sf::IntRect(0, 0, 1, 1) and sf::IntRect(1, 1, 1, 1)
-/// don't intersect.
+/// This means that IntRect({0, 0}, {10, 10}) and IntRect({10, 0}, {10, 10})
+/// will give intersection ({1, 0}, {1, 10})
 ///
-/// sf::Rect is a template and may be used with any numeric type, but
+/// Rect is a template and may be used with any numeric type, but
 /// for simplicity type aliases for the instantiations used by SFML are given:
-/// \li sf::Rect<int> is sf::IntRect
-/// \li sf::Rect<float> is sf::FloatRect
-///
-/// So that you don't have to care about the template syntax.
+/// \li Rect<int> is IntRect
+/// \li Rect<float> is FloatRect
 ///
 /// Usage example:
 /// \code
-/// // Define a rectangle, located at (0, 0) with a size of 20x5
-/// sf::IntRect r1(0, 0, 20, 5);
+/// // Define a rectangle, located at (0, 0) with a size of (10, 20)
+/// IntRect rect1({0, 0}, {10, 20});
 ///
-/// // Define another rectangle, located at (4, 2) with a size of 18x10
-/// sf::Vector2i position(4, 2);
-/// sf::Vector2i size(18, 10);
-/// sf::IntRect r2(position, size);
+/// // Be careful, negative size is allowed!
+/// IntRect rect2({0, 0}, {-10, -20});
 ///
 /// // Test intersections with the point (3, 1)
-/// bool b1 = r1.contains(3, 1); // true
-/// bool b2 = r2.contains(3, 1); // false
+/// bool test1 = rect1.contains(5, 10);  // true
+/// bool test2 = rect1.contains(10, 20); // false
 ///
-/// // Test the intersection between r1 and r2
-/// std::optional<sf::IntRect> result = r1.findIntersection(r2);
+/// // Test the intersection between rect1 and rect2
+/// std::optional<IntRect> result = rect1.findIntersection(rect2);
 /// // result.has_value() == true
-/// // result.value() == (4, 2, 16, 3)
+/// // result.value() == IntRect(0, 0, 1, 1)
```

Comment:
This pseudo-constructor call doesn't exist. Maybe just leave this how it was before.